XamarinCommunityToolkit/Views/SegmentedView/Android/SegmentedViewRenderer.android.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using Android.Content;
+using Xamarin.CommunityToolkit.Android.UI.Views;
+using Xamarin.CommunityToolkit.UI.Views;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+using RadioButton = Android.Widget.RadioButton;
+
+[assembly: ExportRenderer(typeof(SegmentedView), typeof(SegmentedViewRenderer))]
+
+namespace Xamarin.CommunityToolkit.Android.UI.Views
+{
+	public class SegmentedViewRenderer : ViewRenderer<SegmentedView, FormsSegments>
+	{
+		readonly Context context;
+		FormsSegments control;
+
+		public SegmentedViewRenderer(Context context)
+			: base(context)
+		{
+			this.context = context;
+		}
+
+		protected override void OnElementChanged(ElementChangedEventArgs<SegmentedView> e)
+		{
+			base.OnElementChanged(e);
+
+			if (Control == null)
+			{
+				InitializeControl(Element.Items);
+			}
+
+			if (e.OldElement != null && Control != null)
+				InvalidateControl();
+		}
+
+		void InvalidateControl()
+		{
+			Control.SegmentSelected -= SegmentSelected;
+			((INotifyCollectionChanged)Element.Items).CollectionChanged -= SegmentsCollectionChanged;
+		}
+
+		void InitializeControl(IList<string> segments)
+		{
+			control = new FormsSegments(context);
+
+			((INotifyCollectionChanged)Element.Items).CollectionChanged += SegmentsCollectionChanged;
+
+			SetNativeControl(control);
+			PopulateSegments(segments);
+
+			Control.DisplayMode = Element.DisplayMode;
+			Control.CornerRadius = Element.CornerRadius;
+			Control.SegmentSelected += SegmentSelected;
+		}
+
+		void PopulateSegments(IList<string> segments)
+		{
+			for (var i = 0; i < segments.Count; i++)
+			{
+				Control.Children.Add(segments[i]);
+			}
+		}
+
+		void SegmentsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+		{
+			switch (e.Action)
+			{
+				case NotifyCollectionChangedAction.Add:
+					for (var s = 0; s < e.NewItems.Count; s++)
+					{
+						Control.Children.Add(e.NewItems[s].ToString());
+					}
+					break;
+				case NotifyCollectionChangedAction.Remove:
+					for (var s = 0; s < e.OldItems.Count; s++)
+					{
+						Control.Children.RemoveAt(e.OldStartingIndex);
+					}
+					break;
+				case NotifyCollectionChangedAction.Reset:
+				default:
+					Control.Children.Clear();
+					break;
+			}
+		}
+
+		void SegmentSelected(object sender, SelectedPositionChangedEventArgs e)
+		{
+			Element.SelectedIndex = (int)e.SelectedPosition;
+		}
+
+		/// <summary>
+		/// When a property of an element changed.
+		/// </summary>
+		/// <param name="sender">Sender.</param>
+		/// <param name="e">Xamarin.Forms Elements</param>
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			base.OnElementPropertyChanged(sender, e);
+
+			if (Control == null || Element == null)
+				return;
+
+			if (e.PropertyName == SegmentedView.SelectedIndexProperty.PropertyName)
+			{
```

Comment:
we don't use `{ }` for one line code. Can you remove it and others, as you did in line 118?

Reply:
Done.